src/handlers/pinnedMessages.ts

```diff
@@ -0,0 +1,128 @@
+import lineByLine from 'n-readlines'
+import { Entity, entities } from '../Entities'
+import log from '../helpers/logger'
+import { getMapping, getRoomId, getMessageId } from '../helpers/storage'
+import {
+  axios,
+  formatUserSessionOptions,
+} from '../helpers/synapse'
+
+export type PinnedMessages = { [key: string]: string[] }
+export type UserForRoom = { [key: string]: string }
+export type PromiseResult = {
+    pinnedMessages: PinnedMessages;
+    userForRoom: UserForRoom;
+};
+
+/**
+ * Type of Rocket.Chat messages
+ */
+export type RcMessage = {
+  _id: string
+  t?: string // Event type
+  rid: string // The unique id for the room
+  msg: string // The content of the message.
+  tmid?: string
+  ts: {
+    $date: string
+  }
+  mentions?: string[]
+  u: {
+    _id: string
+    username?: string
+    name?: string
+  }
+  // md?: any // The message's content in a markdown format.
+  pinned?: boolean
+  pinnedBy?: {
+    _id?: string
+    username?: string
+  }
+  drid?: string // The direct room id (if belongs to a direct room).
+  // attachments?: any[] // An array of attachment objects, available only when the message has at least one attachment.
+  reactions?: {
+    [key: string]: {
+      usernames: string[]
+    }
+  }
+}
+
+/**
+ * Reads the input file for messages, gets the mappings of each pinned message and returns this collection
+ * It also returns a dict to know which user needs to connect to pinned messages in a room
+ * @returns A collection with: PinnedMessages collection with messages IDs of rooms & Dict of rooms: users
+ */
+export async function getPinnedMessages(): Promise<PromiseResult> {
+  const pinnedMessages: PinnedMessages = {}
+  const userForRoom: UserForRoom = {}
+  const rl = new lineByLine(`./inputs/${entities.messages.filename}`)
+  let line: false | Buffer
+  while ((line = rl.next())) {
+    const message: RcMessage = JSON.parse(line.toString())
+    if (message.pinned) {
+      log.info(message._id)
+      const matrixRoomId = await getRoomId(message.rid)
+      const matrixMessageId = await getMessageId(message._id)
+      if (!matrixRoomId) {
+        log.warn(
+          `Room ${message.rid} has no mapping, thus no pinned messages.`
+        )
+        continue
+      }
+
+      if (message.pinnedBy !== undefined) {
```

Comment:
I'd check for the existence and truthy values like

```suggestion
      if (message.pinnedBy) {
```

as well as in the other checks like this. Or do you do it intentionally like this?

Reply:
no I think your way is fine!